packages/ember-template-compiler/lib/plugins/transform-attrs-into-args.ts

```diff
@@ -64,24 +64,17 @@ export default function transformAttrsIntoArgs(env: EmberASTPluginEnvironment):
         if (isAttrs(node, stack[stack.length - 1])) {
           let path = b.path(node.original.substr(6)) as AST.PathExpression;
 
-          deprecate(
-            `Using {{attrs}} to reference named arguments has been deprecated. {{attrs.${
-              path.original
-            }}} should be updated to {{@${path.original}}}. ${calculateLocationDisplay(
-              moduleName,
-              node.loc
-            )}`,
-            false,
-            {
-              id: 'attrs-arg-access',
-              url: 'https://deprecations.emberjs.com/v3.x/#toc_attrs-arg-access',
-              until: '4.0.0',
-              for: 'ember-source',
-              since: {
-                enabled: '3.26.0',
-              },
-            }
-          );
+          if (node.this === false) {
+            assert(
+              `Using {{attrs}} to reference named arguments has been deprecated. {{attrs.${
```

Comment:
*If* we added an assertion:

```suggestion
              `Using {{attrs}} to reference named arguments is unsupported. {{attrs.${
```

And this test should be renamed to `assert-against-attrs` in the style of https://github.com/emberjs/ember.js/pull/19691/files, it is no longer a transform. The transforming behaviors here should be dropped.

**However** this removal does not need to add an assert against `{{attrs}}`. We can just drop the transform.

The suggestion in my last review was that a template lint could be added to ember-template-lint against `{{this.attrs}}`. Once we remove the deprecation in this PR, `{{this.attrs}}` will Just Work in Ember components like any other property access. We should lint against it, but we cannot *break* it. That would make it special, and our goal is to have fewer special things.

Reply:
It doesn't seem (based on [a comment](https://github.com/emberjs/ember.js/blob/master/packages/ember-template-compiler/lib/plugins/transform-attrs-into-args.ts#L24)) that the transform is not needed. Also, if I comment out the transform, tests that use `this.attrs.foo` fail with

```
Expected: "In layout - someProp: something here"
Result: "In layout - someProp: [object Object]"
```